cmd/eksctl-anywhere/cmd/listpackages.go

```diff
@@ -0,0 +1,130 @@
+
+	"github.com/aws/eks-anywhere-packages/pkg/artifacts"
+	"github.com/aws/eks-anywhere-packages/pkg/bundle"
+	"github.com/aws/eks-anywhere-packages/pkg/testutil"
+	"github.com/aws/eks-anywhere/pkg/curatedpackages"
+	"github.com/aws/eks-anywhere/pkg/dependencies"
+	"github.com/aws/eks-anywhere/pkg/kubeconfig"
+	"github.com/aws/eks-anywhere/pkg/version"
+)
+
+type listPackagesOption struct {
+	kubeVersion string
+	source      curatedpackages.BundleSource
+	registry    string
+	username    string
+	password    string
+}
+
+var lpo = &listPackagesOption{}
+
+func init() {
+	listCmd.AddCommand(listPackagesCommand)
+	listPackagesCommand.Flags().Var(&lpo.source, "source", "Discovery Location. Options (cluster, registry)")
+	err := listPackagesCommand.MarkFlagRequired("source")
+	if err != nil {
+		log.Fatalf("Error marking flag as required: %v", err)
+	}
+	listPackagesCommand.Flags().StringVar(&lpo.kubeVersion, "kubeversion", "", "Kubernetes Version of the cluster to be used. Format <major>.<minor>")
+	listPackagesCommand.Flags().StringVar(&lpo.registry, "registry", "", "Used to specify an alternative registry for discovery")
+	listPackagesCommand.Flags().StringVar(&lpo.username, "username", "", "Username for custom registry provided in registry flag")
+	listPackagesCommand.Flags().StringVar(&lpo.password, "password", "", "Password for custom registry provided in registry flag")
+}
+
+var listPackagesCommand = &cobra.Command{
+	Use:          "packages",
+	Short:        "Lists curated packages available to install",
+	PreRunE:      preRunPackages,
+	SilenceUsage: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if err := validateKubeVersion(lpo.kubeVersion, lpo.source); err != nil {
+			return err
+		}
+		if err := validateRegistryCredentials(lpo.registry, lpo.username, lpo.password); err != nil {
+			return err
+		}
+
+		if err := listPackages(cmd.Context()); err != nil {
+			return err
+		}
+		return nil
+	},
+}
+
+func listPackages(ctx context.Context) error {
+	kubeConfig := kubeconfig.FromEnvironment()
+	deps, err := newDependencies(ctx, kubeConfig)
+	if err != nil {
+		return fmt.Errorf("unable to initialize executables: %v", err)
+	}
+
+	bm := createBundleManager(lpo.kubeVersion)
+	registry, err := newRegistry(ctx, deps)
+	if err != nil {
+		return err
+	}
+
+	b := curatedpackages.Bundle{
+		KubeConfig:    kubeConfig,
+		Source:        lpo.source,
+		KubeVersion:   lpo.kubeVersion,
+		Kubectl:       deps.Kubectl,
+		BundleManager: bm,
+		CliVersion:    version.Get(),
+		Registry:      registry,
+	}
+
+	bundle, err := b.GetLatestBundle(ctx)
+	if err != nil {
+		return err
+	}
+	packages := curatedpackages.Package{
+		Packages: bundle.Spec.Packages,
+	}
+	packages.DisplayPackages()
+	return nil
+}
+
+func createBundleManager(kubeVersion string) bundle.Manager {
+	versionSplit := strings.Split(kubeVersion, ".")
+	if len(versionSplit) != 2 {
+		return nil
+	}
+	major, minor := versionSplit[0], versionSplit[1]
+	log := logr.Discard()
```

Comment:
A comment about using this might clear things up. I assume this returns a client